server/publications/afOrders.js

```diff
@@ -65,3 +65,14 @@ Meteor.publish('searchOrders', function () {
   }
   return this.ready();
 });
+
+Meteor.publish('nonWarehouseOrders', function () {
+  shopId = ReactionCore.getShopId();
+  if (Roles.userIsInRole(this.userId, ['admin', 'owner', 'dashboard/advanced-fulfillment', 'reaction-advanced-fulfillment'], ReactionCore.getShopId())) {
+    return ReactionCore.Collections.Orders.find({
+      'shopId': shopId,
+      'advancedFulfillment.workflow.status': 'nonWarehouseOrder'
+    });
+  }
+  return this.ready();
+});
```

Comment:
I like it!
One thing I noticed looking this over, that I've got in my publications too, is that we don't need to call `ReactionCore.getShopId()` twice. We can just use `shopId` in the `Roles.userIsInRole` function call.